packages/block-editor/src/components/rich-text/native/index.native.js

```diff
@@ -843,9 +877,8 @@ export class RichText extends Component {
 		if ( this.props.value !== this.value ) {
 			this.value = this.props.value;
 		}
-		const { __unstableIsSelected: isSelected } = this.props;
-
 		const { __unstableIsSelected: prevIsSelected } = prevProps;
+		const { __unstableIsSelected: isSelected } = this.props;
 
 		if ( isSelected && ! prevIsSelected ) {
```

Comment:
I understand that the code executed with this condition is also related to focusing the text input after an update. So, I wonder if we should include this logic into the new function `shouldFocusTextInputAfterUpdate`. WDYT?

Reply:
The thing is `shouldFocusTextInputAfterUpdate` only handles merges between two RichTexts so I wonder if we should keep the current code as it is and rename `shouldFocusTextInputAfterUpdate` to `shouldFocusTextInputAfterMerge` or something like that, what do you think?

Ideally `componentDidUpdate` should also be refactored as it currently handles forcing the focus of a TextInput, blur, and font styles 😅 maybe we can't change much there but at least we could split it into different functions to make it easier to read/understand.